File: 1-50/2/main.erl
-module(main).
-export([main/0]).

main() ->
    ANSWER = calc(1, 1, 0),
    io:format("~p~n", [ANSWER]).

calc(A, B, SUM) ->
    if A >= 4000000 -> SUM;
       A rem 2 == 0 -> calc(B, A+B, SUM+A);
       true         -> calc(B, A+B, SUM)
    end.

File: 1-50/2/main.hs
main :: IO ()
main =
  let answer = calc 1 1 0
  in
  putStrLn $ show answer

calc :: Int -> Int -> Int -> Int
calc a b s
  | a >= 4000000 = s
  | otherwise    = if a `mod` 2 == 0 then calc b (a+b) (s+a)
                                     else calc b (a+b) s
